SpokenNumber.js

```diff
@@ -5,26 +5,15 @@ function SpokenNumber(number) {
         throw new Error('Number must be between 0 and 999,999,999,999.');
     }
 
-    this.periods = this.toPeriods(number.toString().split(''));
+    this.periods = this.toPeriods(number.toString());
 }
 
 SpokenNumber.prototype.toPeriods = function (digits) {
 
-    let periods = [];
-    while (digits.length) {
-        periods.push(digits.splice(-3));
-    }
-
-    return periods.map((period, index) => {
-        let suffixes = {
-            0: '',
-            1: 'thousand',
-            2: 'million',
-            3: 'billion'
-        }
+    let periods = digits.replace(/\B(?=(\d{3})+(?!\d))/g, ",").split(',')
```

Comment:
wow, that's some regex there! Could you break it down for me so I know what it's doing?